petaio.c

```diff
@@ -438,25 +438,25 @@ petaio_build_buffer(BigArray * array, IOTableEntry * ent, int * selection, int N
         int tid = omp_get_thread_num();
         int NT = omp_get_num_threads();
         if(NT > All.NumThreads) abort();
-        int start = (selection?((size_t) NumSelection):((size_t)NumPart)) * tid / NT;
-        int end = (selection?((size_t) NumSelection):((size_t)NumPart)) * (tid + 1) / NT;
-        int localsize = 0;
+        int start = (selection ? NumSelection : NumPart) * tid / NT;
+        int end = (selection ? NumSelection : NumPart) * (tid + 1) / NT;
         npartThread[tid] = 0;
         for(i = start; i < end; i ++) {
             int j = selection?selection[i]:i;
             if(P[j].Type != ent->ptype) continue;
             npartThread[tid] ++;
         }
 #pragma omp barrier
-        offsetThread[0] = 0;
```

Comment:
I don't think there is a race -- the crash might be due to something else. 
At any step we can be assure the previous offset is computed.
Your patch makes localsize invalid except the master branch.